framework/generated/dx12_generators/dx12_enum_to_string_header_generator.py

```diff
@@ -69,19 +74,27 @@ def generate_feature(self):
 
         # Generate REFIID handler
         body = 'std::string ToString(const IID& riid);'
-        body += '\ntemplate <> std::string ToString<IID>(const GUID& obj, ToStringFlags toStringFlags, uint32_t tabCount, uint32_t tabSize);'
+        body += '\ninline std::string ToString(const GUID& obj, ToStringFlags toStringFlags, uint32_t tabCount, uint32_t tabSize){ return ToString(obj); }'
         write(body, file=self.outFile)
 
     def write_include(self):
-        code = ''
-        header_dict = self.source_dict['header_dict']
-        for k, v in header_dict.items():
-            code += '#include <{}>\n'.format(k)
-
-        code += '#include "format/platform_types.h"\n'
-        code += '#include "util/defines.h"\n'
-        code += '#include "util/to_string.h"\n'
+        # We don't need everything in self.source_dict['header_dict']:
```

Comment:
I feel like the DXGI header list here should come from DXGI_SOURCE_LIST, otherwise this makes a second place requiring an update when a new DXGI header is released.  Is this change just a performance improvement reducing the number of headers included?

Reply:
Just done to cut down headers pulled-in from memory. Good point about the one definition. I can back out this change.

Reply:
I pushed a fix.